Source/WebCore/dom/TrustedTypeUtils.cpp

```diff
@@ -0,0 +1,162 @@
+    }
+};
+
+String trustedTypeToString(const TrustedType trustedType)
+{
+    switch (trustedType) {
+    case TrustedType::TrustedHTML:
+        return "TrustedHTML"_s;
+    case TrustedType::TrustedScript:
+        return "TrustedScript"_s;
+    case TrustedType::TrustedScriptURL:
+        return "TrustedScriptURL"_s;
+    default:
+        ASSERT_NOT_REACHED();
+        return nullString();
+    }
+}
+
+// https://w3c.github.io/trusted-types/dist/spec/#process-value-with-a-default-policy-algorithm
+std::variant<std::nullptr_t, Exception, Ref<TrustedHTML>, Ref<TrustedScript>, Ref<TrustedScriptURL>> processValueWithDefaultPolicy(ScriptExecutionContext& scriptExecutionContext, TrustedType expectedType, const String& input, const String& sink)
+{
+    RefPtr<TrustedTypePolicy> protectedPolicy;
+    if (RefPtr document = dynamicDowncast<Document>(scriptExecutionContext)) {
+        if (auto window = document->domWindow()) {
+            auto trustedTypesFactory = WindowOrWorkerGlobalScopeTrustedTypes::trustedTypes(*window);
+            protectedPolicy = trustedTypesFactory->defaultPolicy();
+        }
+    } else if (RefPtr workerGlobalScope = dynamicDowncast<WorkerGlobalScope>(scriptExecutionContext)) {
+        auto trustedTypesFactory = WindowOrWorkerGlobalScopeTrustedTypes::trustedTypes(*workerGlobalScope);
+        protectedPolicy = trustedTypesFactory->defaultPolicy();
+    }
+
+    if (!protectedPolicy)
+        return nullptr;
+
+    VM& vm = scriptExecutionContext.vm();
+
+    auto jsExpectedType = JSC::jsString(vm, trustedTypeToString(expectedType));
+    JSC::Strong<JSC::Unknown> strongExpectedType(vm, jsExpectedType);
+    auto jsSink = JSC::jsString(vm, sink);
+    JSC::Strong<JSC::Unknown> strongSink(vm, jsSink);
+    FixedVector<JSC::Strong<JSC::Unknown>> arguments({ strongExpectedType, strongSink });
+    auto policyValueHolder = protectedPolicy->getPolicyValue(expectedType, input, WTFMove(arguments), false);
+    if (policyValueHolder.hasException())
+        return { policyValueHolder.releaseException() };
+
+    auto policyValue = policyValueHolder.releaseReturnValue();
+    if (policyValue.isNull())
+        return nullptr;
+
+    if (expectedType == TrustedType::TrustedHTML)
+        return { TrustedHTML::create(policyValue) };
+    if (expectedType == TrustedType::TrustedScript)
+        return { TrustedScript::create(policyValue) };
+    if (expectedType == TrustedType::TrustedScriptURL)
+        return { TrustedScriptURL::create(policyValue) };
+
+    ASSERT_NOT_REACHED();
+    return nullptr;
+}
+
+// https://w3c.github.io/trusted-types/dist/spec/#get-trusted-type-compliant-string-algorithm
+ExceptionOr<String> getTrustedTypeCompliantString(TrustedType expectedType, ScriptExecutionContext& scriptExecutionContext, const String& input, const String& sink)
+{
+    String stringValue(input);
+
+    if (!scriptExecutionContext.settingsValues().trustedTypesEnabled)
+        return stringValue;
```

Comment:
Might want `WTFMove(stringValue)` for this and the many other cases here.

Reply:
WTFMove here causes build failures on the linux pipelines (WPE and GTK).